src/pyspartn/spartnhelpers.py

```diff
@@ -208,27 +208,31 @@ def escapeall(val: bytes) -> str:
     return "b'{}'".format("".join(f"\\x{b:02x}" for b in val))
 
 
-def convert_timetag(timetag16: int, timetag32: int = None) -> int:
+def convert_timetag(timetag16: int, basedate: datetime = datetime.now()) -> int:
     """
     Convert 16-bit timetag to 32-bit format.
-    16-bit format = half days in seconds
-    32-bit format = total seconds since 2010-01-01
 
-    TODO it appears this may require the 32-bit timetag from an earlier SPARTN message
+    32-bit format represents total seconds since TIMEBASE (2010-01-01).
+    16-bit format represents seconds past nearest half day date.
+
+    e.g. if nearest half day date is "2023-06-27 12:00:00", a timetag16
+    of 32580 represents "2023-06-27 12:00:00" + 12*3600 + 32580,

```

Comment:
shouldn't this be "2023-06-27 00:00:00" + 12*3600 + 32580?

Reply:
done